### PR TITLE
Update Supported LLMs

### PR DESCRIPTION
**Pull request by Dominastorm**

### CODE REVIEW

File: uptrain/operators/language/llm.py [hunk 1]
```diff
@@ -70,9 +74,12 @@ async def async_process_payload(
                             openai.error.APIError,
                             openai.error.Timeout,
                             openai.error.TryAgain,
+                            cohere.error.CohereAPIError,
```
Comment (ishaan-jaff) [inline uptrain/operators/language/llm.py:77]: litellm standardizes exceptions to the openai format, so you don't need to check cohere errors https://docs.litellm.ai/docs/exception_mapping

**Comment by ishaan-jaff** [comment]
@Dominastorm this is an awesome PR ! what's your twitter/linkedin - would love to give you a shoutout

File: uptrain/framework/base.py [hunk 2]
```diff
@@ -25,7 +25,11 @@ class Settings(BaseSettings):
 
     # external api related
     openai_api_key: str = Field(None, env="OPENAI_API_KEY")
-    openai_rpm_limit: int = 100
+    cohere_api_key: str = Field(None, env="COHERE_API_KEY")
+    huggingface_api_key: str = Field(None, env="HUGGINGFACE_API_KEY")
+    openai_rpm_limit: int = 2000
```
Comment (sourabhagr) [inline uptrain/framework/base.py:30]: @Dominastorm

File: uptrain/operators/language/llm.py (hunk 1 above)
Comment (Dominastorm) [reply uptrain/operators/language/llm.py]: Made the change. Thanks!

File: uptrain/operators/language/generation.py [hunk 3]
```diff
@@ -130,7 +130,6 @@ def setup(self, settings: Settings):
 
     def _make_payload(self, id: t.Any, text: str, model: str) -> Payload:
         return Payload(
-            endpoint="chat.completions",
```
Comment (sourabhagr) [inline uptrain/operators/language/generation.py:133]: @Dominastorm - this is fine right?

Comment (Dominastorm) [reply uptrain/operators/language/generation.py:133]: Yes

File: uptrain/operators/language/llm.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 from loguru import logger
 from tqdm.asyncio import tqdm_asyncio
 from pydantic import BaseModel, Field
```
Comment (sourabhagr) [inline uptrain/operators/language/llm.py:14]: @Dominastorm - can we do lazy_load? Check https://github.com/uptrain-ai/uptrain/blob/9b38b06a7ec06f68afd884665d571986c0d1bfc5/uptrain/operators/embs.py#L25C1-L26C1

File: uptrain/operators/language/llm.py [hunk 5]
```diff
@@ -81,6 +82,16 @@ async def async_process_payload(
...
+                    fallback = {
+                        "gpt-3.5-turbo": "gpt-3.5-turbo-16k",
+                        "gpt-3.5-turbo-0613": "gpt-3.5-turbo-16k-0613",
+                        "gpt-4": "gpt-4-32k",
+                        "gpt-4-0613": "gpt-4-32k-0613",
+                    }
+
+                    payload.data["model"] = fallback[payload.data["model"]]
                     # TODO: check the model being used and not always switch to 3.5 16k variant
                     payload.data["model"] = "gpt-3.5-turbo-16k"
```
Comment (sourabhagr) [inline uptrain/operators/language/llm.py:96]: @Dominastorm - I guess this line should be deleted?

File: uptrain/operators/language/llm.py (hunk 4 above)
Comment (Dominastorm) [reply uptrain/operators/language/llm.py]: Done

File: uptrain/operators/language/llm.py (hunk 5 above)
Comment (Dominastorm) [reply uptrain/operators/language/llm.py]: Yeah sorry, missed that. Fixed it now

**Review by sourabhagr — APPROVED** [review]

**Review by ananis25 — COMMENTED** [review]
litellm needs to be added as a dependency for the package

**Pull request merged**